### ISSUE TITLE
The chunk is very big. 

### ISSUE THREAD

**Issue by joseph-vedadi** [react-icons/react-icons#289]
I wonder what is the most updated way to import icons .. 
I only use 10 icons at most and the chunk size is very very big. 

I wonder how could I optimize this . 

```js
import { FiCompass } from 'react-icons/fi'
import { FaSitemap, FaRegCalendarCheck } from 'react-icons/fa'
import { GiProcessor } from 'react-icons/gi'
import { IoMdTrain, IoIosPhonePortrait, IoIosRefresh } from 'react-icons/io'
import { AiOutlineQuestionCircle, AiTwotoneExperiment } from 'react-icons/ai'
import { MdWatch } from 'react-icons/md'
```

**Comment by joseph-vedadi** [comment]
![image](https://user-images.githubusercontent.com/17168374/75848382-ee237480-5d96-11ea-8afc-4871a2476c47.png)

**Comment by joseph-vedadi** [comment]
Any help ?

**Comment by glinkot** [comment]
What actual size was the chunk in kb?

**Comment by joseph-vedadi** [comment]
fa: 1.3 M
ai: 626kb
md: 409kb
fi: 133 kb
![image](https://user-images.githubusercontent.com/17168374/77146958-7ec1ac00-6a49-11ea-8851-ea108da4a859.png)

**Comment by joseph-vedadi** [comment]
any updates to fix this issue

**Comment by claweyenuk** [comment]
The issue is that for any imported icon libraries, all icons are imported.

We haven't dug into this, but ideally, only the icons that are used would be imported. Since we use <10 icons, we are considering putting the SVG's into a small library that we can use. But, it would be great if this isn't necessary to get the file size down.

**Comment by kamijin-fanta** [comment]
Discussed here. By the way, the demo site for this project successfully produces small chunks. If the settings are correct, only the icons used will be imported.

https://github.com/react-icons/react-icons/issues/154

**Comment by Nate-Wessel** [comment]
IMO, this is a pretty critical issue. Has there been any progress? I don't use more than about a dozen icons and my minified bundle is ~200kb. No good.

**Comment by RangerCoder99** [comment]
I just run in the same issue with GI!

**Comment by naingaungphyo** [comment]
+1 
I imported only few icons like OP but I got the long loading time for 4s+ and the size in the network tab is around 10MB.

Here is the import

`// import { MdSubtitles, MdCategory, MdPlace, MdLocationCity, MdComment, MdDeleteForever, MdAddAPhoto } from 'react-icons/md'
// import { IoMdTrain } from 'react-icons/io'
// import { IoBus } from 'react-icons/io5'
// import { FaYenSign } from 'react-icons/fa'
// import { GiMoneyStack } from 'react-icons/gi'`

**Comment by naingaungphyo** [comment]
@kamijin-fanta Hi bro, is there anyway to optimize import.

Thanks

**Comment by SpencerKaiser** [comment]
@naingaungphyo I'm using Next.js and had the same concern. I checked my prod app and the file in question was insignificant in size (9.16kb) but I very much had the issue when running locally in development mode...

Try building your app with `NODE_ENV=production` and see if your file size shrinks like mine did 😄

**Comment by claweyenuk** [comment]
For what it's worth, someone on my team updated our libraries and it looked like this issue went away. I wish I had more data to share but I don't have it off-hand. We moved to webpack 5.44.x and babel 7.14.x

**Comment by SpencerKaiser** [comment]
@claweyenuk good context, but I'm definitely still seeing this issue locally on `4.2.0` - I know I'm a couple of minor releases behind though, so maybe it was fixed in one of those releases.

**Comment by claweyenuk** [comment]
@SpencerKaiser  We're on react-icons 4.2.0 as well. I should probably add react 17 to the list. We updated all our libs at once and the issue want away for us.

Previously we were on react-icons 4.2.0, react 16.13.1, babel 6.26.3, and wepback 4.43.0.

**Comment by codsane** [comment]
@SpencerKaiser I'm seeing this on `4.4.0` as well, so maybe the fix in the parent comment was related to webpack or babel?

**Comment by Crypt0zauruS** [comment]
Hello, react-icons 4.7.1 with Nextjs 13.1.2. Same problem here, analyzer shows clearly a huge chunk due to react icons. No better result with all-files.

**Comment by nephix** [comment]
There is no `@react-icons/all-files` `4.7.1`, the current version is [`4.1.0`](https://www.npmjs.com/package/@react-icons/all-files)

**Comment by Crypt0zauruS** [comment]
Yes, that's why i've uninstalled first `react-icons` ` 4.7.1` to give a try to `@react-icons/all-files` `4.1.0`  But same result, chunk remains at near 10MB . Do i have to clean the cache or something else ?

**Comment by Mattijah** [comment]
This is ridiculous. I'm not even using any from those icons yet this gets included and it makes up half of the size of the entire app.

**Comment by kirkbushell** [comment]
Same issue here - using next js with react icons - file size is like 10mb for using 8 icons lmfao. There's clearly an import issue in the library or libraries!

**Comment by void-hr** [comment]
I am using <GiHamburgerMenu /> in my nextjs component, the chunk file size of that particular page shoots from less than 1 to 7.5MB

**Comment by ddeisadze** [comment]
Any updates?

**Comment by jpmelnik** [comment]
Any updates?

**Comment by agi-omar** [comment]
same issue here - using nextjs 14 - 1.2 mb for like 8 icons

**Comment by HassanMunene** [comment]
Has this issue been resolved still facing the same problem in 2024!

**Comment by joelsena** [comment]
I am using react-icons in my standalone react project using webpack, and in the bundle analyzer it clearly shows that the non-parsed size of the files increased a LOT, thankfully using production optimization it decreases a lot making it actually useful compared to other icon libraries. But in the Development build it just unnecessary BIG (which may impact performance in development).

**Comment by kirkbushell** [comment]
> I am using react-icons in my standalone react project using webpack, and in the bundle analyzer it clearly shows that the non-parsed size of the files increased a LOT, thankfully using production optimization it decreases a lot making it actually useful compared to other icon libraries. But in the Development build it just unnecessary BIG (which may impact performance in development).

Oh it's awful, I removed it completely and just built my own icon library (which takes no time at all).

**Comment by stychu** [comment]
Wooaaaa 2025 and the issue is still present and very critical... Keep off of this package in your dependency unless you want hundrets of kbs being bloated into our app

**Comment by joelsena** [comment]
> > I am using react-icons in my standalone react project using webpack, and in the bundle analyzer it clearly shows that the non-parsed size of the files increased a LOT, thankfully using production optimization it decreases a lot making it actually useful compared to other icon libraries. But in the Development build it just unnecessary BIG (which may impact performance in development).
> 
> Oh it's awful, I removed it completely and just built my own icon library (which takes no time at all).

With further analysis in my app's built bundle with optimizations, webpack handled pretty well react-icons imports letting only the necessary icons into it. Although in the development build react-icons chunk occupies a LOT of space it didn't impact that much and compared to other icon libraries i tested, react-icons gave me more results since it's easy to use and has a variety of icons. 

That being said, i am pretty sure making my own icon library would be better but i have no idea where to start, can you share a little of how you did it? My biggest concerns it's the time to make one and handling the licenses and appropriately references it.

**Comment by S3gillu** [comment]
Is there any update on the same, it's still very big in size in production as well

<img width="722" alt="Image" src="https://github.com/user-attachments/assets/573d27eb-9b96-4d8a-9d93-302a2deac311" />

**Comment by mxfng** [comment]
Still no tree shaking?
